paasta_tools/automatic_rollbacks/state_machine.py

```diff
@@ -96,20 +106,51 @@ def start_timer(self, timeout, trigger, message_verb):
         def times_up():
             self.update_slack_thread(f"Time's up, will now {message_verb}.")
             self.trigger(trigger)
+            self.timer_handle = None
+            self.timer_running = False
 
         def schedule_callback():
             """Unfortunately, call_at is not threadsafe, and there's no call_at_threadsafe, so we need to schedule the
             call to call_at with call_soon_threadsafe."""
             self.timer_handle = self.event_loop.call_later(timeout, times_up)
+            self.timer_trigger = trigger  # This allows cancel_timer to selectively cancel.
+            self.timer_timeout = timeout  # saved for restart_timer
+            self.timer_message_verb = message_verb  # saved for restart_timer
+            # self.timer_started()
```

Comment:
Something else actually starts the timer on the method, right?

Reply:
oh, oops, `timer_started` was meant to deal with an issue that I papered over with the `timer_running` boolean. I'll clean this up.